src/lib.rs

```diff
@@ -101,12 +102,17 @@ pub fn reflink_or_copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> io::Resu
 }
```

Comment:
I think we need to improve doc for `reflink_or_copy` and its implementation.

It fallbacks to [`fs::copy`](https://doc.rust-lang.org/std/fs/fn.copy.html) which can handle file but not symlink or dir.

However it calls `check_is_file_and_error`, so I think it makes sense to move the new error checking logic into `reflink` and keep the old one in `reflink_copy`